static/app/views/replays/detail/memoryPanel/domNodesChart.tsx

```diff
@@ -0,0 +1,184 @@
+            </div>
+            <div class="tooltip-footer" style="border: none;">
+              ${t('Time within replay: %s', formatTime(firstValue.axisValue))}
+            </div>
+          <div class="tooltip-arrow"></div>
+        `;
+      },
+    }),
+    xAxis: XAxis({
+      type: 'time',
+      axisLabel: {
+        formatter: (time: number) => formatTime(time, false),
+      },
+      theme,
+    }),
+    yAxis: YAxis({
+      type: 'value',
+      name: t('DOM Nodes'),
+      theme,
+      nameTextStyle: {
+        padding: [8, 8, 8, 48],
+        fontSize: theme.fontSizeLarge,
+        fontWeight: 600,
+        lineHeight: 1.2,
+        fontFamily: theme.text.family,
+        color: theme.gray300,
+      },
+      minInterval: 100,
+      maxInterval: Math.pow(1024, 4),
+      axisLabel: {
+        formatter: (value: number) => axisLabelFormatter(value, 'number', true),
+      },
+    }),
+    onMouseOver: ({data}) => {
+      if (data[0]) {
+        setCurrentHoverTime(data[0]);
+      }
+    },
+    onMouseOut: () => {
+      setCurrentHoverTime(undefined);
+    },
+    onClick: ({data}) => {
+      if (data.value) {
+        setCurrentTime(data.value);
+      }
+    },
+  };
+
+  const staticSeries: Series[] = useMemo(
+    () => [
+      {
+        id: 'nodeCount',
+        seriesName: t('Number of DOM nodes'),
+        data: datapoints.map(d => ({
+          value: d.count,
+          name: d.endTimestampMs - startTimestampMs,
+        })),
+        stack: 'dom-nodes',
+        lineStyle: {opacity: 0, width: 2},
+      },
+      {
+        id: 'replayStart',
+        seriesName: 'Replay Start',
+        data: [{value: 0, name: 0}],
+        lineStyle: {opacity: 0, width: 0},
+      },
+      {
+        id: 'replayEnd',
+        seriesName: 'Replay End',
+        data: [{value: 0, name: durationMs}],
+        lineStyle: {opacity: 0, width: 0},
+      },
+    ],
+    [datapoints, durationMs, startTimestampMs]
+  );
+
+  const currentTimeSeries = useMemo(
+    (): Series => ({
+      id: 'currentTime',
+      seriesName: t('Current player time'),
+      data: [],
+      markLine: {
+        symbol: ['', ''],
+        data: [{xAxis: currentTime}],
+        label: {show: false},
+        lineStyle: {type: 'solid', color: theme.purple300, width: 2},
+      },
+    }),
+
+    [currentTime, theme.purple300]
+  );
+
+  const hoverTimeSeries = useMemo(
+    (): Series => ({
+      id: 'hoverTime',
+      seriesName: t('Hover player time'),
+      data: [],
+      markLine: {
+        symbol: ['', ''],
+        data: currentHoverTime ? [{xAxis: currentHoverTime}] : [],
+        label: {show: false},
+        lineStyle: {type: 'solid', color: theme.purple200, width: 2},
+      },
+    }),
+    [currentHoverTime, theme.purple200]
+  );
+
+  return (
+    <AreaChart
+      series={staticSeries.concat(currentTimeSeries, hoverTimeSeries)}
```

Comment:
Does `useMemo` do here since we're always creating a new array?

Reply:
yeah, i could wrap this in a useMemo too. I was thinking that hoverTime changes all the time, but that's not true when you hover over the chart. The click/hover targets are really small inside the chart, so if you're interacting with it it's actually really stable props wise.